### PR TITLE
Move collector and rule-evaluator to yaml

### PR DESCRIPTION
**Pull request by pintohutch**
Same as https://github.com/GoogleCloudPlatform/prometheus-engine/pull/227 but CLA covers myself as the contributor.

### CODE REVIEW

File: cmd/operator/deploy/operator/10-collector.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      - name: storage
+        emptyDir: {}
+      - name: config
+        configMap:
+          name: collector
+      - name: config-out
+        emptyDir: {}
+      - name: collection-secret
+        secret:
+          secretName: collection
```
Comment (damemi) [inline cmd/operator/deploy/operator/10-collector.yaml:161]: nit: no newline at end of file

File: pkg/operator/collection.go [hunk 2]
```diff
@@ -198,259 +193,55 @@ func (r *collectionReconciler) ensureCollectorSecrets(ctx context.Context, spec
...
-				PriorityClassName:            r.opts.PriorityClass,
-				SecurityContext:              podSpecSecurityContext(),
-			},
-		},
-	}
-	// DNS policy should be set explicitly when using hostNetwork.
-	if r.opts.HostNetwork {
-		ds.Template.Spec.HostNetwork = true
-		ds.Template.Spec.DNSPolicy = "ClusterFirstWithHostNet"
-	}
```
Comment (damemi) [inline pkg/operator/collection.go:445]: don't see these in the new collector manifest, just checking that's intentional

File: cmd/operator/deploy/operator/10-collector.yaml (hunk 1 above)
Comment (pintohutch) [reply cmd/operator/deploy/operator/10-collector.yaml:161]: Done.

File: pkg/operator/collection.go (hunk 2 above)
Comment (pintohutch) [reply pkg/operator/collection.go:445]: Yeah - we don't have a reason to run on the host network anymore. We used to provide the option via the operator, but with collectors being broken out to their own manifests, users can add that themselves if they need it.

**Pull request merged**